### PR TITLE
refactor: switch osmomath to sdkmath; create aliases for future imports; rename BigDec API

### PR DESCRIPTION
**Pull request by p0mvn**
Progress towards: https://github.com/osmosis-labs/osmosis/issues/6099

## What is the purpose of the change

This PR proposes creating an alias for `sdkmath` types in `osmomath`.

Primarily, its goal is to remove the need for the verbose and unreadable prefix "Legacy" from `Dec`.

To avoid having to directly import both `sdkmath` and `osmomath` in files where both `Dec` and `Int`/`Uint` types are used we also create aliases for integer types. The aliases are created in the new file `sdk_math_alias.go`

To avoid name clashes with `BigDec` and `BigDec`, their constrictor APIs are now broken to contain "Big" in the suffix. For example, `NewDec` -> `NewBigDec`.

Limiting this PR to `osmomath` only to seek initial acceptance of the change by the team before proceeding to change the entire repo.

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [x] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

File: osmomath/sdk_math_alias.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	MinInt            = sdkmath.MinInt
+	MaxInt            = sdkmath.MaxInt
+
+	// Uint
+	NewUint = sdkmath.NewUint
+)
+
+func ToDec(i Int) Dec {
+	return i.ToLegacyDec()
+}
```
Comment (p0mvn) [inline osmomath/sdk_math_alias.go:55]: Linking the new alias

File: osmomath/decimal.go [hunk 2]
```diff
@@ -75,64 +75,64 @@ func precisionInt() *big.Int {
 	return new(big.Int).Set(precisionReuse)
 }
 
-func ZeroDec() BigDec     { return BigDec{new(big.Int).Set(zeroInt)} }
-func OneDec() BigDec      { return BigDec{precisionInt()} }
-func SmallestDec() BigDec { return BigDec{new(big.Int).Set(oneInt)} }
+func ZeroBigDec() BigDec     { return BigDec{new(big.Int).Set(zeroInt)} }
+func OneBigDec() BigDec      { return BigDec{precisionInt()} }
+func SmallestBigDec() BigDec { return BigDec{new(big.Int).Set(oneInt)} }
```
Comment (p0mvn) [inline osmomath/decimal.go:80]: Linking the BigDec API break

File: go.work [hunk 3]
```diff
@@ -13,3 +13,5 @@ use ./tests/cl-genesis-positions
 use .
 
 use ./x/epochs
+
+replace github.com/cosmos/cosmos-sdk => ../cosmos-sdk
```
Comment (PaddyMc) [inline go.work:17]: Is this supposed to be here?

**Review by PaddyMc — COMMENTED** [review]
LGTM :zap: 

Is there a timeline on when the sdk is going to re implement their non-legacy math functions? 

Is the plan to remove these aliases when the sdk releases?

**Pull request closed without merging**